modules/core/src/main/java/org/apache/ignite/internal/processors/cache/CacheAffinitySharedManager.java

```diff
@@ -2920,6 +2921,46 @@ private CacheGroupNoAffOrFilteredHolder createHolder(
         return new CacheGroupNoAffOrFilteredHolder(ccfg.getRebalanceMode() != NONE, cctx, aff, initAff);
     }
 
+    /**
+     * Prints {@code waitInfo} to provided {@code log}.
+     *
+     * @param log Logger to print to.
+     */
+    public void printWaitInfo(IgniteLogger log) {
```

Comment:
I think the good practice is catching an exception if it would be thrown during collect the string.
By the way, partition map is not thread safe, hence it can throw CME (partsMap#forEach).